### PR TITLE
lint: plugins + remove year from header

### PR DESCRIPTION
**Pull request by JadenSimon**
Just some lint stuff:
* Remove year from header
* Add special project-specific lint plugins (found in `plugins` in project root, this directory may be used for other plugins, not just lint-related)
  * One rule checks for AWS in localize calls
  * The other is no mocha arrows (may just use the Mocha plugin but this is light-weight)
* Fix violations of the above

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1890,6 +1891,7 @@
         "esbuild-loader": "2.13.1",
         "eslint": "^7.28.0",
         "eslint-config-prettier": "8.3",
+        "eslint-plugin-aws": "file:plugins/eslint-aws",
```
Comment (justinmk3) [inline package.json:1894]: let's mention this in https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#tooling

```
### AWS eslint plugin

Defines custom lint rules for the project.
```

**Comment by JadenSimon** [comment]
Need to change GitHub CI jobs to fix. Will do in separate PR later.

File: .eslintignore [hunk 2]
```diff
@@ -3,4 +3,6 @@ third-party
 **/*.json
 src/shared/telemetry/clienttelemetry.d.ts
 **/*.gen.ts
+plugins/**
 src/testFixtures/workspaceFolder/ts-plain-sam-app/
+.eslintrc.js
```
Comment (justinmk3) [inline .eslintignore:8]: why not lint `plugins/` or this file?

Comment (JadenSimon) [reply .eslintignore:8]: I disabled plugins temporarily but guess I forgot to add it back.

For `.eslintrc.js` there's a really weird linter error that simply does not go away even on mainline. Will add it back when I can figure out what it is.

File: plugins/eslint-aws/lib/rules/no-mocha-arrows.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    if (node.type !== 'CallExpression') {
+        return undefined
+    }
+
+    if (node.callee.type === 'Identifier' && MOCHA_EXPRESSIONS.has(node.callee.name)) {
+        return <any>node.arguments.find(arg => arg.type === 'ArrowFunctionExpression')
+    }
+}
+
+// https://eslint.org/docs/developer-guide/working-with-rules
```
Comment (justinmk3) [inline plugins/eslint-aws/lib/rules/no-mocha-arrows.ts:22]: url was redirected:

```suggestion
// https://eslint.org/docs/latest/extend/custom-rules
```

surprised that `ESTree` is not mentioned in this PR diff, I guess there's no typescript definitions for it?

**Comment by justinmk3** [comment]
Implemented custom lint rules in: https://github.com/aws/aws-toolkit-vscode/pull/4118

These rules from this PR are tracked in https://github.com/aws/aws-toolkit-vscode/issues/2347 for future reference:

- `plugins/eslint-aws/lib/rules/no-localize-aws.ts`
- `plugins/eslint-aws/lib/rules/no-mocha-arrows.ts`

**Pull request closed without merging**